spec/index.md

```diff
@@ -2340,6 +2340,38 @@ This function returns fee percentiles, measured in millisatoshi/vbyte (1000 mill
 
 The [standard nearest-rank estimation method](https://en.wikipedia.org/wiki/Percentile#The_nearest-rank_method), inclusive, with the addition of a 0th percentile is used. Concretely, for any i from 1 to 100, the ith percentile is the fee with rank `⌈i * 100⌉`. The 0th percentile is defined as the smallest fee (excluding coinbase transactions).
 
+### IC method `take_snapshot` {#ic-take_snapshot}
+
+This method takes a snapshot of the canister. Subsequent `take_snapshot` calls will replace the existing snapshot with a new one.
+
+It's important to note that a snapshot will increase the memory footprint of the canister. Thus, the balance must have the right amount of cycles to support the new freezing threshold.
+
+Only controllers can take a snapshot of a canister and restore it.
+
+:::note
+
+It's important to stop a canister before taking a snapshot to ensure that all outstanding callbacks are completed. Failing to do so may cause the canister to not make sense of the callbacks if its state is restored using the snapshot.
+It is expected that the canister admin (or their tooling) does this separately.
+
+:::
+
+### IC method `load_snapshot` {#ic-take_snapshot}
+
+This method loads a snapshot identified by `snapshot_id` onto the canister.  This method fails if the snapshot was previously deleted.
```

Comment:
Is it recommended to try and stop the canister before loading the snapshot? Are there any downsides to loading a snapshot on a non-stopped canister?

Reply:
> Is it recommended to try and stop the canister before loading the snapshot?
Yes, it should be.

> Are there any downsides to loading a snapshot on a non-stopped canister?
The loaded snapshot might not know how to deal with outstanding callbacks.

Reply:
So basically, the requests and responses that are in progress will continue to flow as expected, they will not be touched by the snapshots. However, the canister state can change arbitrarily, so it's up to the developer to decide when to take a snapshot and to restore it. 
 It's a behavior similar to installing code. Loading a snapshot will update the canister module, so again, the canister might not be able to make sense of the callbacks.